protocol/requests/community_metrics_request.go

```diff
@@ -0,0 +1,44 @@
+package requests
+
+import (
+	"errors"
+
+	"github.com/status-im/status-go/eth-node/types"
+)
+
+var ErrNoCommunityID = errors.New("community metrics request has no community id")
+var ErrInvalidTimestampIntervals = errors.New("community metrics request invalid time intervals")
+
+type CommunityMetricsRequestType uint
+
+const (
+	CommunityMetricsRequestMessagesTimestamps CommunityMetricsRequestType = iota + 1
+	CommunityMetricsRequestMessagesCount      CommunityMetricsRequestType = iota + 1
+	CommunityMetricsRequestMembers
+	CommunityMetricsRequestControlNodeUptime
+)
+
+type MetricsIntervalRequest struct {
+	StartTimestamp uint64 `json:"startTimestamp"`
+	EndTimestamp   uint64 `json:"endTimestamp"`
+}
+
+type CommunityMetricsRequest struct {
+	CommunityID types.HexBytes              `json:"communityId"`
+	Type        CommunityMetricsRequestType `json:"type"`
+	Intervals   []MetricsIntervalRequest    `json:"intervals"`
+}
+
+func (r *CommunityMetricsRequest) Validate() error {
```

Comment:
We could have some tests for the validations, but it's fine without since it's fairly simple.

Reply:
added small test case checking invalid request intervals